paperqa/docs.py

```diff
@@ -306,22 +308,32 @@ async def aadd(  # noqa: PLR0912
 
         # try to extract DOI / title from the citation
         if (doi is title is None) and parse_config.use_doc_details:
+            # TODO: specify a JSON schema here when many LLM providers support this
             result = await llm_model.run_prompt(
                 prompt=parse_config.structured_citation_prompt,
                 data={"citation": citation},
                 skip_system=True,
             )
-            with contextlib.suppress(json.JSONDecodeError):
-                clean_text = result.text.strip("`")
-                if clean_text.startswith("json"):
-                    clean_text = clean_text.replace("json", "", 1)
+            clean_text = result.text.strip("`")
+            if clean_text.startswith("json"):
+                clean_text = clean_text.replace("json", "", 1)
+            try:
                 citation_json = json.loads(clean_text)
                 if citation_title := citation_json.get("title"):
                     title = citation_title
                 if citation_doi := citation_json.get("doi"):
                     doi = citation_doi
                 if citation_author := citation_json.get("authors"):
                     authors = citation_author
+            except (json.JSONDecodeError, AttributeError):
+                # json.JSONDecodeError: clean_text was not actually JSON
+                # AttributeError: citation_json was not a dict (e.g. a list)
+                logger.exception(
```

Comment:
I think we need to consider some of the other modes here that aren't really failures. I think it will be really common (a majority of the cases) that only title and authors can be extracted. In those cases, I don't think we want to show this warning. 

I think we should only show this if it fails on both title and doi, otherwise, if either is present, we should continue without warning the user.

Reply:
So if we hit a:
- JSON decode error: we are guaranteed to miss all of title, authors, and DOI
- AttributeError (due to `list` confusion): we are guaranteed to miss all of title, authors, and DOI

Partial failures (as you're trying to protect) can only happen when we successfully decoded JSON and its output was actually a `dict`

Reply:
What I did was both add a comment, and adjust the logged warning to reflect this design detail